src/boot/efi/pe.c

```diff
@@ -209,45 +209,33 @@ static uint32_t get_compatibility_entry_address(const DosFileHeader *dos, const
         return 0;
 }
 
-EFI_STATUS pe_kernel_info(
-                const void *base,
-                uint32_t *ret_entry_point_address,
-                uint32_t *ret_size_of_image,
-                uint32_t *ret_section_alignment) {
-
-        const DosFileHeader *dos;
-        const PeFileHeader *pe;
-
+EFI_STATUS pe_kernel_info(const void *base, uint32_t *ret_compat_address) {
         assert(base);
-        assert(ret_entry_point_address);
+        assert(ret_compat_address);
 
-        dos = (const DosFileHeader *) base;
+        const DosFileHeader *dos = (const DosFileHeader *) base;
         if (!verify_dos(dos))
                 return EFI_LOAD_ERROR;
 
-        pe = (const PeFileHeader *) ((const uint8_t *) base + dos->ExeHeader);
+        const PeFileHeader *pe = (const PeFileHeader *) ((const uint8_t *) base + dos->ExeHeader);
         if (!verify_pe(pe, /* allow_compatibility= */ true))
                 return EFI_LOAD_ERROR;
 
         /* Support for LINUX_INITRD_MEDIA_GUID was added in kernel stub 1.0. */
         if (pe->OptionalHeader.MajorImageVersion < 1)
                 return EFI_UNSUPPORTED;
 
-        uint32_t entry_address = pe->OptionalHeader.AddressOfEntryPoint;
```

Comment:
What about this value, are we dropping use of this entirely?

Reply:
That's the idea. If we use StartImage we don't have to find an entry point. (We'd only want a compat entry point if one exists for efi mixed mode).

Reply:
Makes sense